backend/src/services/toyServices.ts

```diff
@@ -23,14 +27,90 @@ export const fetchToyById = async (id: string) => {
   }
 };
 
-export const fetchAllToys = async (page: number, limit: number) => {
+export const fetchAllToys = async (req: Request) => {
```

Comment:
Consider refactoring for better readability and separation of concerns.
`export const fetchAllToys = async (req: Request) => {
  try {
    const queryObj = { ...req.query };
    const excludeFields = [...EXCLUDED_QUERY_FIELDS];
    excludeFields.forEach(el => delete queryObj[el]);

    let query = Toy.find(queryObj);

    const filters = {
      category: (req.query["category"] as string) || "",
      ageCategory: (req.query["ageCategory"] as string) || "",
      brand: (req.query["brand"] as string) || ""
    };

    Object.keys(filters).forEach(filter => {
      if (filters[filter]) {
        const filterArr = filters[filter].split(",").map(el => new RegExp(el, "i"));
        query = query.find({ [filter]: { $in: filterArr } });
      }
    });

    if (req.query["fields"]) {
      query = query.select((req.query["fields"] as string).split(",").join(" "));
    } else {
      query = query.select("-__v");
    }

    if (req.query["search"]) {
      const search = new RegExp(req.query["search"] as string, "i");
      query = query.find({
        $or: [{ name: search }, { description: search }, { brand: search }]
      });
    }

    if (req.query["sort"]) {
      query = query.sort((req.query["sort"] as string).split(",").join(" "));
    } else {
      query = query.sort("-createdAt");
    }

    const total = await Toy.countDocuments(query);
    const page = Number(req.query["page"]) || PAGINATION.page;
    const limit = Number(req.query["limit"]) || PAGINATION.limit;
    query = query.skip((page - 1) * limit).limit(limit);

    const toys = await query;
    if (!toys?.length) return undefined;

    logger.info(`Toys fetched successfully ${toys}`);
    return { toys, total, page, limit };
  } catch (err) {
    logger.error(`Failed to fetch toys: ${err.message}`);
    throw new Error((err as Error).message);
  }
};
`